maintenance/health/servicehealthcheck/healthchecker_test.go

```diff
@@ -48,30 +52,97 @@ func setup(t *testHealthChecker) *http.Response {
 }
 
 func TestHandlerOK(t *testing.T) {
-	resp := setup(&testHealthChecker{name: "test"})
+	resp := setupOneHealthCheck(&testHealthChecker{name: "test"})
 	if resp.StatusCode != 200 {
 		t.Errorf("Expected /health to respond with 200, got: %d", resp.StatusCode)
 	}
 	helperCheckResponse(t, "OK\n", resp)
 }
 
 func TestHandlerInitErr(t *testing.T) {
-	resp := setup(&testHealthChecker{name: "TestHandlerInitErr", initErr: true})
+	resp := setupOneHealthCheck(&testHealthChecker{name: "TestHandlerInitErr", initErr: true})
 	if resp.StatusCode != 503 {
 		t.Errorf("Expected /health to respond with 503, got: %d", resp.StatusCode)
 	}
 	helperCheckResponse(t, "ERR", resp)
 }
 
 func TestHandlerHealthCheckErr(t *testing.T) {
-	resp := setup(&testHealthChecker{name: "TestHandlerHealthCheckErr", healthCheckErr: true})
+	resp := setupOneHealthCheck(&testHealthChecker{name: "TestHandlerHealthCheckErr", healthCheckErr: true})
 	if resp.StatusCode != 503 {
 		t.Errorf("Expected /health to respond with 503, got: %d", resp.StatusCode)
 	}
 	helperCheckResponse(t, "ERR", resp)
 
 }
 
+func TestHealthCheckAllWithErrors(t *testing.T) {
+
+	hcs := []*testHealthChecker{
+		{name: "TestHandlerHealthCheckErr", healthCheckErr: true},
+		{name: "TestHandlerInitErr", initErr: true},
+		{name: "test"},
+	}
+	expected := []string{
+		"TestHandlerHealthCheckErr : ERR",
+		"TestHandlerInitErr : ERR",
+		"test : OK",
+	}
+
+	checkAllTestHcs(hcs, t, expected, 503)
+}
+func TestHealthCheckAllSuccess(t *testing.T) {
+
+	hcs := []*testHealthChecker{
+		{name: "test2"},
+		{name: "test3"},
+		{name: "test"},
+	}
+	expected := []string{
+		"test3 : OK",
+		"test2 : OK",
+		"test : OK",
+	}
+
+	checkAllTestHcs(hcs, t, expected, 200)
+}
+
+func checkAllTestHcs(hcs []*testHealthChecker, t *testing.T, expected []string, statusCode int) {
+	checks = sync.Map{}
+	initErrors = sync.Map{}
+	rec := httptest.NewRecorder()
+	for _, hc := range hcs {
+		RegisterHealthCheck(hc, hc.Name())
+	}
+	req := httptest.NewRequest("GET", "/health/all", nil)
+	Handler().ServeHTTP(rec, req)
+	resp := rec.Result()
+	defer resp.Body.Close()
+	if resp.StatusCode != statusCode {
+		t.Errorf("Expected /health to respond with %d, got: %d", statusCode, resp.StatusCode)
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	checkRes := strings.Split(string(data), "\n ")
+	if exp := len(checkRes); exp != 4 {
+		t.Errorf("Expected 3 HealthCheck results, but got %v", len(hcs)-1)
```

Comment:
It's probably more future-proof to use len(expected) as count here instead of 3.